albumentations/augmentations/mixing/transforms.py

```diff
@@ -0,0 +1,116 @@
+    image: Union[str, Path]
+    mask: NotRequired[np.ndarray]
+    class_label: NotRequired[np.ndarray]
+
+
+class MixUp(DualTransform):
+    """MixUp data augmentation for images, masks, and class labels.
+
+    This transformation performs the MixUp augmentation by blending an input image, mask, and class label
+    with another set from a predefined reference dataset. The blending is controlled by a parameter lambda,
+    sampled from a Beta distribution, dictating the proportion of the mix between the original and reference data.
+
+    The MixUp augmentation is known for improving model generalization by encouraging linear behavior between
+    classes and smoothing the decision boundaries. It can be applied not only to the images but also to the
+    segmentation masks and class labels, providing a comprehensive data augmentation strategy.
+
+    Reference:
+        Zhang, H., Cisse, M., Dauphin, Y.N., and Lopez-Paz, D., mixup: Beyond Empirical Risk Minimization,
+        ICLR 2018. https://arxiv.org/abs/1710.09412
+
+    Args:
+    ----
+        reference_data Sequence[ReferenceImage]: A sequence of dictionaries containing the reference
+            images, masks, and class labels for mixing. Each dictionary should have 'image', but 'mask', and
+            'class_label' are optional. Defaults to an empty list, resulting in no operation if not provided.
+        read_fn Callable[[Any], Dict[str, np.ndarray]]: A function to load and process the data
+            from the reference_data dictionaries. It should accept one argument (one of the dictionaries) and
+            return a processed dictionary containing the keys 'image' and optionally 'mask', and 'class_label'
+            with their corresponding np.ndarray values. Class label should be one-hot encoded.
+            Defaults to a lambda function that returns its input.
+        alpha (float, optional): The alpha parameter of the Beta distribution used to sample the lambda value.
+            Must be greater than or equal to 0. Higher values make the distribution closer to uniform,
+            resulting in more balanced mixing. Defaults to 0.4.
+        always_apply (bool, optional): Whether the transform should always be applied, regardless of the
+            probability `p`. Defaults to False.
+        p (float, optional): Probability that the transform will be applied. Defaults to 0.5.
+
+    Raises:
+    ------
+        ValueError: If the alpha parameter is negative.
+
+    Notes:
+    -----
+        - If no reference data is provided, this transform will issue a warning and act as a no-op.
+        - The lambda value used for mixing is sampled once per batch and applied globally across all images
+          and labels within that batch.
+        - The class labels are expected to be numpy arrays, enabling the mixing process when applicable.
+
+    """
+
+    def __init__(
+        self,
+        reference_data: Sequence[ReferenceImage] = [],
```

Comment:
I think better to use some generator to get images. Some users might want to take provide image and some additional data into this generator to generate reference image.